### ISSUE TITLE
Can't change the title of exported notes

### ISSUE THREAD

**Issue by macrospecter** [windingwind/zotero-better-notes#542]
This plug-in is amazing! I just have what's hopefully a simple question: I can't figure out how to change the naming convention for MD files when I export them from Zotero. A "save" window pops up as soon as I click export, taking me to the folder I use for my Zotero MD files, but there's no option to change the title. And what I end up with is an awkward auto title for the file that's something like the first 12 words of the note plus the note number. So I'm wondering (a) is there a way to change the file naming convention? and (b) if not, is it possible to view and edit the file name for a given MD file before I export? (I'm hoping for the former because I may want to batch export a bunch of notes.) Thank you!

**Comment by windingwind** [comment]
The `[ExportMDFileName]` template controls this.

**Comment by macrospecter** [comment]
Thanks so much! I’m sorry, the “file name” part of the template name was not showing in my template editor because of window size. One more question: Could you tell me if there’s a way to add to the title a number indicating the number of the note relative to the parent item? Like “1” and “2” if I have two notes? I didn’t see that in the syntax guide.

**Comment by windingwind** [comment]
Replace the template content in editor:

```js
${(noteItem.getNoteTitle ? noteItem.getNoteTitle().replace(/[/\\?%*:|"<> ]/g, "-") + "-" : "")}${noteItem.key}${
(()=>{
if(noteItem.parentItem){
return "-"+noteItem.parentItem.getNotes().length
}
else{
return ""
}
})()
}.md
```

**Comment by macrospecter** [comment]
Hmm that doesn't seem to work. I get this error:

SyntaxError: expected expression, got ')'

**Comment by windingwind** [comment]
Well, how do you use it? This is not used for import. This should be pasted to the template `[ExportMDFileNameV2]` (or without V2, or in the future V higher, depending on your plugin version. You will find the template name not editable and that's it) and it will be used when syncing your markdown.

**Comment by macrospecter** [comment]
I actually wasn't ever able to do this. Do you have any suggestions?

Here is the error I kept getting when I tried to substitute the template you gave for the file name.

**Comment by macrospecter** [comment]
Sorry, here's the error: Template [ExportMDFileNameV2] Error: SyntaxError: expected expression, got ')'

**Comment by macrospecter** [comment]
And to clarify, I can get this portion of the code to work, if I add it as a separate piece in the filename template: {noteItem.parentItem.getNotes().length} else { return "" }}

I just can't get this whole larger portion (which I assume determines whether there is more than one note?) to work without the syntax error:

{
(()=&gt;{
if(noteItem.parentItem){
return "-"+noteItem.parentItem.getNotes().length
}
else{
return ""
}
}

**Comment by windingwind** [comment]
Seem like what you showed above is not the same as what I provided above and there is a syntax problem with your `whole larger portion`. It's obvious not what I provided. You might not be using the right template content.

**Comment by macrospecter** [comment]
Wow, I'm sorry. It works now. Somehow every time I copy the template from email (where I also received your message), the ">" character gets converted to "&gt;." No idea why! I must've been copying the template from email each time that I tried it in Zotero. Thanks for your patience. 

For anyone else who might be interested in modifying file names, here's the final template I arrived at based on windingwind's core template. It does the same as the above but uses the citekey rather than the note name. 

${(noteItem.parentItem
      ? noteItem.parentItem.getField("citekey")
      : "")}${
(()=>{
if(noteItem.parentItem){
return "-"+noteItem.parentItem.getNotes().length
}
else{
return ""
}
})()
}.md
